swfo/brdf_fallback.py

```diff
@@ -708,46 +704,26 @@ def __get_clean_data(data_filename, data_key, data_window):
     return data_convolved
 
 
-def generate_fallback_metadata_doc(h5info: Dict, doy: str):
-    """
-    Generates a metadata doc for a brdf fallback dataset attributes are derived
-    from the metadata of the source dataset; assume metadata reference is ga-stac-like
-    :param h5info:
-        A dictionary of day-of-year strings to h5 collection references
-    :param doy:
-        A string of the day of year being averaged for the tile
-
-    :return:
-        A metadata dictionary describing the dataset
-    """
-    def _reader(filepath: str, offset='/METADATA/CURRENT'):
-        with h5py.File(filepath, 'r') as src:
-            ds = YAML.load(src[offset][()].item())
-        return ds
-
-    metadata_doc = {
-        'id': str(uuid.uuid4()),  # Not sure what the params would be for deterministic uuid
-        'product': {'href': FALLBACK_PRODUCT_HREF},
-        'crs': None,
-        'geometry': None,  # Need to validate imagery intersection
-        'grids': None,  # Need to verify quality mask
-        'measurements': None,  # TODO
-        'lineage': {
-            'brdf_threshold': [],
-            'doy_average': [],
-        },
-        'properties': {}
-    }
-    start_ds, *_, end_ds = sorted(h5info.keys())
-
-    for datestr, fp in h5info.items():
-        src_md = _reader(fp)
-        curr_doy = datetime.datetime.strptime(datestr, '%Y.%m.%d').strftime('%j')
-        metadata_doc['lineage']['brdf_threshold'].append(src_md['id'])
-        if curr_doy == doy:
-            metadata_doc['lineage']['doy_average'].append(src_md['id'])
-
-        if not metadata_doc.get('grids', None):
+def munge_metadata(fp, start_ds=None, end_ds=None, only_id=True): 
+    """
+    extracts metadata from MCD43A1 .h5 files. Returns only uuid of the 
+    h5 files if only_id is set to True. Else general metadata attributes 
+    of the h5 file is returned. 
+    """
+    with h5py.File(fp, 'r') as src: 
+        src_md = YAML.load(src[METADATA_OFFSET][()].item())
+        
+        if only_id: 
+            return src_md['id']
+        else:
+            metadata_doc = { 
+                'id': str(uuid.uuid4()),  # Not sure what the params would be for deterministic uuid
+                'product': {'href': FALLBACK_PRODUCT_HREF},
+                'lineage': {
+                'brdf_threshold': [],
+                'doy_average': []
```

Comment:
I understand the reasoning behind two sets of provenance now.
We might need to use alternate names to "brdf_threshold" or "doy_average".  We'll discuss.

Reply:
I stuck those names in there without any real analysis of what would be descriptive; probably warrants a discussion.